contracts/StablecoinConverter.sol

```diff
@@ -241,32 +258,36 @@ contract StablecoinConverter is EpochTokenLocker {
         previousSolution.tokenIdsForPrice = tokenIdsForPrice;
     }
 
-    function undoPreviousSolution(uint batchIndex) internal {
-        if (previousSolution.batchId == batchIndex) {
-            for (uint i = 0; i < previousSolution.trades.length; i++) {
-                address owner = previousSolution.trades[i].owner;
-                uint orderId = previousSolution.trades[i].orderId;
-                Order memory order = orders[owner][orderId];
-                uint sellVolume = previousSolution.trades[i].volume;
-                addBalance(owner, tokenIdToAddressMap(order.sellToken), sellVolume);
-            }
-            for (uint i = 0; i < previousSolution.trades.length; i++) {
-                address owner = previousSolution.trades[i].owner;
-                uint orderId = previousSolution.trades[i].orderId;
-                Order memory order = orders[owner][orderId];
-                uint128 sellVolume = previousSolution.trades[i].volume;
-                uint128 buyVolume = getExecutedBuyAmount(
-                    sellVolume,
-                    currentPrices[order.buyToken],
-                    currentPrices[order.sellToken]
-                );
-                revertRemainingOrder(owner, orderId, sellVolume);
-                subtractBalance(owner, tokenIdToAddressMap(order.buyToken), buyVolume);
-            }
+    function undoPreviousSolution() internal {
+        for (uint i = 0; i < previousSolution.trades.length; i++) {
+            address owner = previousSolution.trades[i].owner;
+            uint orderId = previousSolution.trades[i].orderId;
+            Order memory order = orders[owner][orderId];
+            uint sellVolume = previousSolution.trades[i].volume;
+            addBalance(owner, tokenIdToAddressMap(order.sellToken), sellVolume);
         }
+        for (uint i = 0; i < previousSolution.trades.length; i++) {
+            address owner = previousSolution.trades[i].owner;
+            uint orderId = previousSolution.trades[i].orderId;
+            Order memory order = orders[owner][orderId];
+            uint128 sellVolume = previousSolution.trades[i].volume;
+            uint128 buyVolume = getExecutedBuyAmount(
+                sellVolume,
+                currentPrices[order.buyToken],
+                currentPrices[order.sellToken]
+            );
+            revertRemainingOrder(owner, orderId, sellVolume);
+            subtractBalance(owner, tokenIdToAddressMap(order.buyToken), buyVolume);
+        }
+    }
+
+    function checkForBestSolutionSubmission(int256 fee) private {
```

Comment:
Not a big fan of this helper method. It does both a check and a side-effect which is not very obvious from the name. Maybe checkAndOverrideObjectiveValue?